Be able to specify the log Level when system property is not found

batmat opened the pull request:
Some properties are more important than others. One may for example want to log a `WARNING` instead of the current default `CONFIG` when some property can not be found. 

@reviewbybees

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

alvarolobato commented on core/src/main/java/jenkins/util/SystemProperties.java:
```diff
@@ -146,25 +146,43 @@ public static String getString(String key) {
         if (value != null) {
-            if (LOGGER.isLoggable(Level.CONFIG)) {
-                LOGGER.log(Level.CONFIG, "Property (system): {0} => {1}", new Object[] {key, value});
+            if (LOGGER.isLoggable(logLevelIfAbsent)) {
```
In this case should be `logLevel`, you are logging when you find the property not when it is missing.

batmat replied:
Right. Fixed. Switch back to simply logLevel for every places, since anyway ...IfAbsent is not the whole picture: the log will also be used for impossible conversions, not just when it's not specified and so on.

batmat committed:
Be able to specify the log Level when system property is not found or invalid

Some properties are more important than others. One may for example
 want to log a WARNING instead of the current default CONFIG
when some property can not be found, or has an invalid format
(like a string, when asking for an Integer).

daniel-beck commented:
Will there be a followup PR with actual application? So far I have a difficult time imagining when this would be useful.

batmat commented:
@daniel-beck it's linked above already: https://github.com/jenkinsci/jenkins/issues/2645

oleg-nenashev commented on core/src/main/java/jenkins/util/SystemProperties.java:
```diff
@@ -146,25 +146,43 @@ public static String getString(String key) {
+     *
+     * @exception  NullPointerException if {@code key} is {@code null}.
+     * @exception  IllegalArgumentException if {@code key} is empty.
+     */
```
`@since`

oleg-nenashev commented on core/src/main/java/jenkins/util/SystemProperties.java:
```diff
@@ -241,6 +259,24 @@ public static Integer getInteger(String name) {
+     *          If the property is missing, return the default value.
+     *          Result may be {@code null} only if the default value is {@code null}.
+     */
+    public static Integer getInteger(String name, Integer def) {
```
`@since`

oleg-nenashev reviewed (approved):
Looks good to me

oleg-nenashev commented:
🐝 and @reviewbybees done

oleg-nenashev commented:
CC @jenkinsci/code-reviewers

alvarolobato commented:
🐝
